src/test/scala/com/microsoft/hyperspace/index/HybridScanSuite.scala

```diff
@@ -72,175 +74,276 @@ trait HybridScanSuite extends QueryTest with HyperspaceSuite {
       val destPath = new Path(inputFiles(i) + ".copy")
       fs.copyToLocalFile(sourcePath, destPath)
     }
+
     for (i <- 1 to deleteCnt) {
       fs.delete(new Path(inputFiles(inputFiles.length - i)), false)
     }
+
+    val df2 = spark.read.format(sourceFileFormat).load(sourcePath)
+    (df2.inputFiles diff inputFiles, inputFiles diff df2.inputFiles)
   }
 
   before {
+    spark.conf.set(IndexConstants.INDEX_LINEAGE_ENABLED, "true")
     spark.enableHyperspace()
   }
 
   after {
+    FileUtils.delete(systemPath)
     spark.disableHyperspace()
   }
 
-  override def afterAll(): Unit = {
-    super.afterAll()
-    FileUtils.delete(new Path(sampleDataLocationRoot))
+  private def getLatestStableLog(indexName: String): IndexLogEntry = {
+    val entry = logManager(systemPath, indexName).getLatestStableLog()
+    assert(entry.isDefined)
+    assert(entry.get.isInstanceOf[IndexLogEntry])
+    entry.get.asInstanceOf[IndexLogEntry]
   }
 
-  test(
-    "Append-only: join rule, appended data should be shuffled with indexed columns " +
-      "and merged by BucketUnion.") {
-    val df1 = spark.read.format(fileFormat).load(sampleDataFormatAppend)
-    val df2 = spark.read.format(fileFormat).load(sampleDataFormatAppend2)
-    def joinQuery(): DataFrame = {
-      val query = df1.filter(df1("clicks") >= 2000).select(df1("clicks"), df1("query"))
-      val query2 = df2.filter(df2("clicks") <= 4000).select(df2("clicks"), df2("Date"))
-      query.join(query2, "clicks")
-    }
-    val baseQuery = joinQuery()
-    val basePlan = baseQuery.queryExecution.optimizedPlan
-
-    withSQLConf("spark.sql.autoBroadcastJoinThreshold" -> "-1") {
-      withSQLConf(IndexConstants.INDEX_HYBRID_SCAN_ENABLED -> "false") {
-        val join = joinQuery()
-        assert(basePlan.equals(join.queryExecution.optimizedPlan))
+  // Check if the given plan is transformed correctly to handle deleted files. The plan
+  // should include only one LogicalRelation node which is transformed with the given index.
+  def checkDeletedFiles(
```

Comment:
Note to reviewers: Instead of checking the transformed plan in every test, I introduced following functions to check the transformed plan properly.

```scala
def checkFilterIndexHybridScanUnion( // for filter rule
def checkJoinIndexHybridScan( // for join rule
def checkDeletedFiles( // check the deleted files for both filter & join 
```

HybridScanSuite contains general testcases for all non-partitioned/partitioned/delta lake dataset.
Dataset-specific tests are located in each test class.